src/OrbitGl/ThreadStateBar.cpp

```diff
@@ -209,6 +212,23 @@ std::string ThreadStateBar::GetThreadStateSliceTooltip(PrimitiveAssembler& primi
   return tooltip;
 }
 
+void ThreadStateBar::DrawThreadStateSliceOutline(PrimitiveAssembler& primitive_assembler,
+                                                 const ThreadStateSliceInfo& slice,
+                                                 const Color& outline_color) const {
+  float left_x = timeline_info_->GetWorldFromTick(slice.begin_timestamp_ns());
+  float right_x = timeline_info_->GetWorldFromTick(slice.end_timestamp_ns());
+  float top_y = GetPos()[1];
+  float bottom_y = GetPos()[1] + GetHeight();
+  std::array<Vec2, 4> outline_points{
+      Vec2{left_x, top_y},
+      Vec2{right_x, top_y},
+      Vec2{right_x, bottom_y},
+      Vec2{left_x, bottom_y},
+  };
+  Quad outline{outline_points};
+  primitive_assembler.AddQuadBorder(outline, GlCanvas::kZValueTimeBar, outline_color);
```

Comment:
Only the TimeBar should be at this value. If you have strong reasons to not choose the same than the non-transparent arrow, consider creating a new constant.

Reply:
I made a new z value for the arrow and changed the z value for the outline. Can you check it now?